src/components/Directions/DirectionsBox.tsx

```diff
@@ -21,24 +23,23 @@ const Wrapper = styled(Stack)<{ $isMobileMode: boolean }>`
 
 export const DirectionsBox = () => {
   const isMobileMode = useMobileMode();
-  const [result, setResult] = useState<RoutingResult>(null);
   const [revealed, revealForm, hide] = useBoolState(false); // mobile only
   const hideForm = isMobileMode && result && !revealed;
 
+  const { setResult } = useDirectionsContext();
+
   const setResultAndHide = useCallback(
     (result: RoutingResult) => {
       setResult(result);
       hide();
     },
```

Comment:
Looks like, you have imported `result` from `lodash` instead of `DirectionContext`. Could you please fix it?